static/js/place/dev_place_main.tsx

```diff
@@ -0,0 +1,499 @@
+import {
+  defaultDataCommonsClient,
+  defaultDataCommonsWebClient,
+} from "../utils/data_commons_client";
+
+/**
+ * Converts the API response from getPlaceCharts into a SubjectPageConfig object.
+ * Groups charts by category and creates the necessary configuration objects for
+ * rendering the subject page.
+ *
+ * @param placeChartsApiResponse The API response containing chart data
+ * @returns A SubjectPageConfig object with categories, tiles, and stat var specs
+ */
+function placeChartsApiResponsesToPageConfig(
+  placeChartsApiResponse: PlaceChartsApiResponse
+): SubjectPageConfig {
+  const chartsByCategory = _.groupBy(
+    placeChartsApiResponse.charts,
+    (item) => item.category
+  );
+  const categoryConfig: CategoryConfig[] = Object.keys(chartsByCategory).map(
+    (categoryName) => {
+      const charts = chartsByCategory[categoryName];
+
+      const tiles: TileConfig[] = charts.map((chart) => {
+        return {
+          description: chart.description,
+          title: chart.title,
+          type: chart.type,
+          statVarKey: chart.statisticalVariableDcids,
+        };
+      });
+
+      const statVarSpec: Record<string, StatVarSpec> = {};
+      charts.forEach((chart) => {
+        chart.statisticalVariableDcids.forEach((variableDcid, variableIdx) => {
+          const denom =
+            chart.denominator &&
+            chart.denominator.length === chart.statisticalVariableDcids.length
+              ? chart.denominator[variableIdx]
+              : undefined;
+          statVarSpec[variableDcid] = {
+            denom,
+            log: false,
+            scaling: chart.scaling,
+            statVar: variableDcid,
+            unit: chart.unit,
+          };
+        });
+      });
+
+      // Group tiles into pairs to show a two-column layout
+      const column1Tiles: TileConfig[] = [];
+      const column2Tiles: TileConfig[] = [];
+      tiles.forEach((tile, index) => {
+        if (index % 2 === 0) {
+          column1Tiles.push(tile);
+        } else {
+          column2Tiles.push(tile);
+        }
+      });
+      const category: CategoryConfig = {
+        blocks: [
+          {
+            columns: [{ tiles }],
+          },
+        ],
+        statVarSpec,
+        title: categoryName,
+      };
+      return category;
+    }
+  );
+
+  const pageConfig: SubjectPageConfig = {
+    metadata: undefined,
+    categories: categoryConfig,
+  };
+  return pageConfig;
+}
+
+/**
+ * Component that renders the header section of a place page.
+ * Displays the place name, category (if not Overview), and subheader text.
+ * Also shows the place DCID with a link to view it in the Knowledge Graph browser.
+ *
+ * @param props.category The current category being viewed
+ * @param props.place The place object containing name and DCID
+ * @param props.placeSubheader HTML string with additional place context
+ * @returns Header component for the place page
+ */
+const PlaceHeader = (props: {
```

Comment:
would be nice if some of these components are moved into separate files

Reply:
I'll address this in a follow-up. I like to have some of the smaller components in the same file, but agree this file is also getting pretty large.